external-crates/move/crates/move-compiler/src/parser/syntax.rs

```diff
@@ -1779,6 +1775,21 @@ fn parse_binop_exp(context: &mut Context, lhs: Exp, min_prec: u32) -> Result<Exp
         context.tokens.advance()?;
         let op_end_loc = context.tokens.previous_end_loc();
 
+        if op_token == Tok::As {
+            let ty = {
+                // numeric types don't have type arguments, so don't attempt to parse '<' as a
+                // type arg
+                let tn = parse_name_access_chain(context, || "a type name")?;
+                sp(tn.loc, Type_::Apply(Box::new(tn), vec![]))
+            };
```

Comment:
Toward parser resiliency, we should check for and discard type params if they're here (or preserve them and complain about the ill-typed cast later).

Reply:
The problem is actual cases like 
```
x as u64 < y
```
We either have to make this whitespace sensitive or just parse a type name. I am happy to do either